UnityAds/UnityRouter.m

```diff
@@ -97,26 +97,14 @@ - (void) setIfUnityAdsCollectsPersonalInfo
 
 - (void)requestVideoAdWithGameId:(NSString *)gameId placementId:(NSString *)placementId delegate:(id<UnityRouterDelegate>)delegate;
 {
-    //Call load first, to minimize reporting discrepencies
-    [UnityAds load:placementId];
-    
-    if([UnityAds getPlacementState:placementId] == kUnityAdsPlacementStateNoFill){
-        NSError *error = [NSError errorWithDomain:MoPubRewardedVideoAdsSDKDomain code:MPRewardedVideoAdErrorNoAdsAvailable userInfo:nil];
-        [delegate unityAdsDidFailWithError:error];
-        return;
-    }
-    
     if (!self.isAdPlaying) {
         [self.delegateMap setObject:delegate forKey:placementId];
         
         if (![UnityAds isInitialized]) {
             [self initializeWithGameId:gameId];
         }
-
-        // Need to check immediately as an ad may be cached.
-        if ([UnityAds isReady:placementId]) {
-            [self unityAdsReady:placementId];
-        }
+        [UnityAds load:placementId];
```

Comment:
I think we should leave this at the top of the function. Because an ad playing really shouldn't affect the ability to load an ad.